local/network.go

```diff
@@ -66,8 +66,18 @@ type localNetwork struct {
 	newAPIClientF api.NewAPIClientF
 	// Used to create new node processes
 	nodeProcessCreator NodeProcessCreator
-	// Closed when network is done shutting down
-	closedOnStopCh chan struct{}
+	stopOnce           sync.Once
+	// Cancelled when Stop begins.
+	onStopCtx context.Context
+	// Cancels [onStopCtx]
+	onStopCtxCancel func()
+	// True if Stop has returned.
+	// This field only exists so that in removeNode,
+	// we know whether Stop has already completed, in
+	// which case we return an error, or not, in which
+	// case we proceed to remove all nodes.
+	// [lock] must be held while accessing this field.
+	stopped bool
```

Comment:
This can be avoided is the check for function preconditions (currently mainly stopped network), is moved to the public methods, and so the non public ones can be used at discretion. What do you think?

Reply:
removed